examples/cmd/README.md

```diff
@@ -0,0 +1,99 @@
+# Command Tool Examples
+
+Learn how to execute system commands in your Roast workflows using the Cmd tool. This example demonstrates how to define allowed commands and use them as functions within your workflows.
+
+## Overview
+
+The `Cmd` tool allows you to define a list of approved system commands that can be executed by the AI. Each allowed command is treated as a distinct function, providing several benefits:
+
+- **Security**: Only explicitly allowed commands can be run, preventing unauthorized system access.
+- **Clarity**: Each command has a clear, defined purpose within the workflow.
+- **Intelligent Usage**: Custom descriptions can be provided to help the AI choose the correct command for a given task.
+
+## Configuration
+
+You can configure the `Cmd` tool in your workflow YAML file.
+
+### Basic Configuration
+
+List the commands that the AI is allowed to execute:
+
+```yaml
+tools:
+  - Roast::Tools::Cmd:
+      allowed_commands:
+        - pwd
+        - ls
+        - echo
+```
+
+Each command listed here (`pwd`, `ls`, `echo`) becomes a function the AI can call. For example, the AI can call `pwd()` to get the current directory, or `ls(args: "-la")` to list files.
+
+### Enhanced Configuration with Descriptions
+
+For more complex commands or to provide better guidance to the AI, you can include a description:
+
+```yaml
+tools:
+  - Roast::Tools::Cmd:
+      allowed_commands:
+        - pwd
+        - ls
+        - name: git
+          description: "git CLI - version control system (e.g., git status, git log)"
+        - name: npm
+          description: "npm CLI - Node.js package manager (e.g., npm install, npm run)"
+        - name: docker
+          description: "Docker CLI - container platform (e.g., docker ps, docker run)"
+```
+These descriptions help the AI understand the purpose of each command and its common subcommands or arguments.
+
+## How It Works
+
+When the `Cmd` tool is configured, each entry in `allowed_commands` is registered as a function available to the AI.
+
+- `pwd` becomes `pwd()`
+- `ls` becomes `ls(args: <string>)`
```

Comment:
@obie as this is, it IS a breaking change because `cmd(` is removed. I can think of a couple ways around that, including tying this functionality to whether `allowed_commands` is provided. I mentioned this when we talked but want to call it out here.